### DISCUSSION TITLE
JS Debugger doesn't seem to work correctly or is it me?

### DISCUSSION THREAD

**Discussion by Chris2011** [apache/netbeans#7785] [Q&A] [answered]
I tested in NetBeans 22 but with a lot of custom configs and today I also tried the new NetBeans 23 with clean user dir.

@matthiasblaesing I dunno whether it is me who not understands the new JS debugger, that you implemented couple of weeks ago or it is not working as expected, that's why created a dicussion over a real ticket.

When I have a nodeJS project (HTML5/JS project with enabled nodejs support) with a main,js file and this piece of code:

```js
const foo = function () {
  console.log("bar");
};

foo();
```

I set a breakpoint at line two - console.log.
The debug protocol is default but I also changed to Chrome DevTools Protocol. After call the action "Debug file", the debug session started, is still running, but it runs the script and says done. Please see my screencapture for more info. Did I miss smth?

Also it seems that the breakpoint in the js file is always wrong.

https://github.com/user-attachments/assets/89aeb92b-b051-4fa7-a8f1-f82c93e44244

In the IDE log I can see this: `INFO [org.netbeans.ui.metrics.debugger]: Debugger session Chrome DevTools Endpoint at ws://localhost:56513/3028aa6f-8872-4319-9b22-e195884ff99e started at null for JavaScript`

**Comment by matthiasblaesing** [comment]
What node version is this? I tested with a (more or less) current master build and node 18.19.1 (the one bundled with Ubuntu 24.04). The breakpoints were correctly set and I halted at the right points. It is correct, that after your code is finished the process is kept running, as this is nodes behavior. The process ends when the debugger disconnects.

**Reply by Chris2011** [reply]
Sry I forgot to mention it. I use 18.16.1. I installed it via NVS.

![image](https://github.com/user-attachments/assets/0f5ccbc5-062e-4c42-b95e-c5cde1bf20de).

Also I use Windows 11 64.
The behaviour I can understand, the debugger is always running. But NetBeans can understand, that the debugger is ended, because it ends as you can see in the last message, but this is a different thing I know that VS Code ends automatically, but needs inventigation.

Let's focus first on this breakpoint thing.

**Reply by matthiasblaesing** [reply, marked as answer]
It is reproducible on windows.

**Reply by Chris2011** [reply]
Thx, I will create a bug for this.

**Reply by Chris2011** [reply]
https://github.com/apache/netbeans/issues/7797, thx for testing
